### PR TITLE
Add 305 release notes

### LINKED ISSUE #342: Release notes for 305
## Dain Sundstrom       
- [ ] all checked
-  03-04 22:24:17    9f66272174 Migrate Parquet dictionary test cases to presto-parquet
-  03-04 22:24:17    df7a2bac87 Support dictionary encoding in Parquet V2
## David Phillips       
- [x] all checked
-  02-27 23:21:23    48eba72958 Use safe concurrent access to local Maven repository
-  02-28 14:38:13    ddca399ced Use correct artifact name for Takari local repository
-  02-28 14:41:19    7693635812 Add Hive file format test for SequenceFile
-  02-28 16:05:02    d072edc8be Fix regular expression failure for Joni
-  03-01 13:16:33    6d93feba22 Cleanup warnings in Hive file format testers
-  03-01 17:49:37    d9f33376c2 Remove legacy Hive config for DFS domain socket
-  03-02 15:34:21    1dfc357424 Remove legacy writers for ORC and RCFile
-  03-02 15:34:21    50a4720526 Allow writing ORC files compatible with Hive 2.x
-  03-02 15:34:21    9f0a9ab9a2 Remove support for DWRF file format
-  03-04 12:13:09    073cf05710 Remove unnecessary tests for null constructor arguments
-  03-04 12:13:09    2830e88dc1 Fix Hive bucket calculation for timestamps with milliseconds
-  03-04 12:13:09    89e1c85330 Extract method for computing Hive bucket number
-  03-04 12:13:09    95bed72b2c Record expected value for Hive bucket hash code
-  03-04 12:13:09    d59cbb6bec Simplify test code for computing Hive bucket number
-  03-04 13:48:44    3611eeb9dd Fix format in TestCsvPrinter
-  03-04 13:48:44    3d322f756a Use enum for CLI CSV output format
-  03-04 13:48:44    e92b9a3db5 Add CSV output option without quotes
-  03-04 18:43:09    02e408c8b5 Fix single backquotes in documentation
-  03-04 18:43:09    ada6d8a3f7 Fail documentation build on single backquote
-  03-05 21:34:06    9d3975f5b2 Update to Parquet 1.10.1
-  03-05 21:34:06    efaecdaf89 Remove unused profile in Parquet POM
-  03-06 20:26:18    4a8665c474 Update CACHE_KEY when setting credential in HDFS configuration
## Grzegorz Kokosiński 
- [x] all checked
-   03-01 00:10:43    98566710d3 Flush Elasticsearch testcase data loading to resolve data inconsistency issue
-   03-01 02:15:17    f05827eddf Pass JdbcIdentity to DriverConnectionFactory#openConnection
-   03-05 00:56:36    6de6020451 Do not ignore failures from cassandra tests
-   03-05 11:42:28    199f85ee11 Accept Optional in BaseJdbcClient#getTableNames
-   03-05 11:42:28    3d343739ed Annotate with @Nullable methods in ConnectorMetadata
-   03-05 11:42:28    5a63e2ec0b Annotate nullable paramters of BaseJdbcClient#getTables
-   03-05 11:42:28    5aa422f2af Return Optional from BaseJdbcClient#getTableHandle
-   03-05 11:42:28    dfc7634042 Remove unnecessary checkArgument
## Karol Sobczak        
- [ ] all checked
-  02-27 23:54:41    6b9c0891c2 Fix RIGHT/OUTER Join stream ordering property derivations
-  02-27 23:55:20    6e0f7d945d Replace internal network rows/data stats in UI
-  02-27 23:55:59    da5051c9fd Remove unused path variable
-  03-01 02:28:36    b1be7a41e6 Add ScanFilterAndProjectOperator benchmark
-  03-01 02:48:11    04c66cd209 Display total current and peak memory in query list view
-  03-01 02:48:11    426af7ee2f Add revocable memory to total reported memory
-  03-01 02:48:11    8f083fadc5 Add peak revocable memory stats to OperatorStats
-  03-01 02:48:11    a29900c0f0 Display peak revocable memory in Query details page
-  03-01 02:48:11    b3d50dfe46 Propagate revocable memory stats to QueryInfo
-  03-01 03:03:52    ab3e533d8c Improve join-max-broadcast-table-size description
-  03-01 03:03:52    d83e8a1ce8 Unhide join_max_broadcast_table_size session property
-  03-01 13:01:10    85ffbd3c6a Remove unused ThriftHiveMetastoreClient constructor
-  03-04 06:50:10    4644cd516c Do not spill for join over table scan
-  03-04 06:50:10    7465c8c296 Add spillable property to JoinNode
-  03-04 06:50:10    c31287fefc Fix query failure for enabled spill and Index Join above Table Scan
-  03-04 06:50:10    e3a1e0c77c Extract join queries tests to separate class
## Martin Traverso      
- [x] all checked
-  03-01 17:01:26    7c346e723b Improve method naming
-  03-01 17:01:26    e42e060a7c Remove unused method
-  03-06 18:28:04    3a1bd58d85 Fix typo
## Piotr Findeisen      
- [x] all checked
-  02-27 12:40:43    72ec55a278 Remove useless call
-  02-27 12:40:43    7d0f46c46b Make numeric casts strict
-  02-27 12:41:29    105afce39e Remove redundant overload
-  02-27 12:41:29    8a900f615b Remove redundant comments
-  02-27 12:41:29    9c957d9e3b Reuse existing padding logic in CharType#getObjectValue
-  02-27 12:41:59    fd18ecda27 Add SortNode stats rule
-  02-27 13:45:20    042d17b4d9 Calculate query peak memory usage as memory cost
-  02-27 13:45:20    1a3626de22 Allow testing recursive cost calculations
-  02-27 13:45:20    3b5fb34354 Use local cost concept for exchange cost estimation
-  02-27 13:45:20    49e5b45527 Remove useless comment
-  02-27 13:45:20    7b80ee573d Use local cost concept in cost calculator using exchanges
-  02-27 13:45:20    bc585f2b94 Remove unused method
-  02-27 13:45:20    bf1afe3abb Rename "cumulative cost" to "cost"
-  02-27 13:45:20    c7f12750ff Add memory cost estimation test for join above accumulation
-  02-27 13:45:20    e438e1ef8b Rename PlanNodeCostEstimate to PlanCostEstimate
-  02-27 13:45:20    fe5cc0e1d4 Make CostCalculator calculate "cumulative cost"
-  02-28 09:24:40    42c337529d Do not use historic dates in SQL Server tests
-  02-28 09:24:40    683b70040b Enable product tests with SQL Server
-  02-28 09:24:40    ecbe96afd8 Update SQL Server docker image
-  03-01 22:45:19    a6d9f2c5f8 Update tempto to 163
-  03-01 23:25:28    f0ed245893 Do not abbreviate variable names
-  03-02 02:02:55    f57e412c03 Move Selectors ending comment to Selectors section
-  03-02 02:10:04    e3488e8d2d Set ClassLoader when calling EventListenerFactory
-  03-04 00:31:31    9a31d12511 Simplify ClassLoader management in Plugins
-  03-04 01:28:41    c3939aadd2 Fix a typo in docs
-  03-04 08:39:02    e5adac45cf Use simpler for loop
-  03-04 13:47:13    14e9d97271 Rename resourceGroups package to resourcegroups
-  03-05 03:04:18    1dd65a9ac3 Make test FileResourceGroupConfigurationManager use API
-  03-05 03:04:18    2d242324ee Refactor test code for readability
-  03-05 04:17:06    063b79b245 Improve validation in MongoClientConfig
-  03-05 05:48:06    311d8f21a0 Refactor testExtractVariableConfiguration()
-  03-06 01:17:30    835d6273bd Log location of etc directory
-  03-06 04:39:12    aca0c5e9ba Fix resource groups configuration example
-  03-06 04:39:12    ce82afcf02 Extract resource groups configuration example to file
-  03-06 06:07:41    358f88e37a Improve LocalMemoryManager tests
-  03-06 06:07:41    5e1f5ae5fa Add tests for LocalMemoryManager constructor
-  03-06 06:07:41    7d5003891b Remove duplicate test
-  03-06 06:27:37    ce9ecf97ed Remove useless output from EXPLAIN
-  03-06 14:00:45    18d7595ab4 Fix logging of etc directory location

### PR DESCRIPTION
**Pull request by electrum**
Fixes #342

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-305.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+General Changes
+---------------
+
+* Fix failure of :doc:`/functions/regexp` for certain patterns and inputs
+  when using the default ``JONI`` library. (:issue:`350`)
+* Fix a rare ``ClassLoader`` related problem for plugins providing an ``EventListenerFactory``. (:issue:`299`)
+* Un-hide ``join_max_broadcast_table_size`` session property. (:issue:`346`)
+* Improve performance of queries when spill is enabled, but spill is not triggered. (:issue:`315`)
+* Improve memory cost estimations by calculating a plan's estimated peak memory usage. (:issue:`247`)
```
Comment (dain) [inline presto-docs/src/main/sphinx/release/release-305.rst]:
```
Consider estimated query peak memory when making cost based decisions.
```

File: presto-docs/src/main/sphinx/release/release-305.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+* Include revocable memory in total memory stats. (:issue:`273`)
+* Add peak revocable memory to operator stats. (:issue:`273`)
+* Add :func:`ST_Points` function to access vertices of a linestring. (:issue:`316`)
+* Add a system table ``system.metadata.analyze_properties``
+  to list all :doc:`/sql/analyze` properties. (:issue:`376`)
+
+Resource Groups Changes
+-----------------------
+
+* Fix selection when selector uses pattern named capturing group as a variable. (:issue:`373`)
```
Comment (dain) [inline presto-docs/src/main/sphinx/release/release-305.rst]: I'm not sure what that means?

File: presto-docs/src/main/sphinx/release/release-305.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Web UI Changes
+--------------
+
+* Display peak revocable memory, and current and peak total memory, in detailed query view. (:issue:`273`)
```
Comment (dain) [inline presto-docs/src/main/sphinx/release/release-305.rst]: Display peak revocable memory,  current total memory, and peak total memory, in detailed query view.

Comment (dain) [reply presto-docs/src/main/sphinx/release/release-305.rst]: I'd still make this change, but it is up to you.

File: presto-docs/src/main/sphinx/release/release-305.rst [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+Release 305
+===========
+
+General Changes
+---------------
+
+* Fix failure of :doc:`/functions/regexp` for certain patterns and inputs
+  when using the default ``JONI`` library. (:issue:`350`)
+* Fix a rare ``ClassLoader`` related problem for plugins providing an ``EventListenerFactory``. (:issue:`299`)
+* Un-hide ``join_max_broadcast_table_size`` session property. (:issue:`346`)
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-305.rst]: "Expose [...], which was previously hidden."

File: presto-docs/src/main/sphinx/release/release-305.rst [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+Resource Groups Changes
+-----------------------
+
+* Fix resource group selection when selector uses regular expression variables. (:issue:`373`)
+
+Web UI Changes
+--------------
+
+* Display peak revocable memory, current total memory,
+  and peak total memory, in detailed query view. (:issue:`273`)
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-305.rst]: The comma before "in" sounds funny.

File: presto-docs/src/main/sphinx/release/release-305.rst [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+* Add option to output CSV without quotes. (:issue:`319`)
+
+Hive Connector Changes
+----------------------
+
+* Fix handling of updated credentials for Google Cloud Storage (GCS). (:issue:`398`)
+* Fix calculation of bucket number for timestamps that contain a non-zero
+  milliseconds value. Previously, data would be written into the wrong bucket,
+  or could be incorrectly skipped on read. (:issue:`366`)
+* Allow writing ORC files compatible with Hive 2.0.0 to 2.2.0 by pretending
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-305.rst]: Why is this relevant: "by pretending that the writer is an old version of Hive" ?

Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-305.rst]: Because it works by identifying itself as Hive in the file

**Pull request merged**